src/rt/ds/stackarray.h

```diff
@@ -0,0 +1,63 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+#pragma once
+
+#include <cassert>
+#include <cstddef>
+
+/**
+ * @brief A stack allocated array.
+ *
+ * Due to portability allocates a largeish array on the stack, if this array is
+ * not big enough then dynamically allocates something of the correct size. This
+ * is done to avoid the need for a dynamic allocation in the common case.
+ */
+template<typename T>
+class StackArray
+{
+  // Size of the stack allocated array.
+  static constexpr size_t Size = 128;
+
+  // Stack allocated array.
+  T main[Size];
+
+  // Pointer to the array in use, may be main or a dynamically allocated array.
```

Comment:
Can we have just one extra line of comment highlighting ownership in case this is dynamically allocated?